Feat: set hostname properly

mbologna opened the pull request.

mbologna committed:
Feat: set hostname properly

moio commented on pillar/config.sls:
```diff
@@ -1 +1,3 @@
 local_domain: moio
+hostname: rpi
```
Default hostname should be raspberrypi

moio commented on pillar/config.sls:
```diff
@@ -1 +1,3 @@
 local_domain: moio
+hostname: rpi
+{%- set dnsdomainname = "michelebologna.net" %}
```
Why is this needed and why is it in set form?

mbologna replied:
This is in a `set` form to use dnsdomainname in other variables (e.g. https://github.com/moio/raspberry-openvpn-gateway/pull/17/commits/c53ad618bb4362d8286f427c6db0c9f8301a9d97)

moio replied:
Please try using a plain pillar variable, thanks!

mbologna replied:
Done!

mbologna committed:
Change default hostname to raspberrypi

mbologna committed:
Fix: use pillar var for domain name

moio committed:
config.sls: make coherent with init.sls

moio committed:
Use only one variable for the domain name

moio committed:
Set /etc/hosts contents more correctly

- use 127.0.1.1 for self-lookups (does not break if network is down)
- use file.replace to avoid duplicated lines over several state
applications

moio committed:
Refactoring: split dnsmasq config files

Do not ship entire configuration file, only relevant differences.

moio committed:
Bugfix: set dnsmasq hosts file separately

See https://hugoheden.wordpress.com/2009/02/24/dnsmasq-and-etchosts/
